### PR TITLE
dual currency

### PR DESCRIPTION
**Pull request by ETeissonniere**
A few things to note for testing:
- we send transfers with `tiimeBalances`
- accounts and balances are in `tiimeAccountStore.account`
- to create some coins you can trigger the extrinsic call `tiimeBalances.setBalance`
- when viewing the chain state with PolkaJS it will show the number along with the CAPS unit, that's expected

**Please make sure to test locally with polka js**

### CODE REVIEW

File: pallets/account-store/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    pub struct Pallet<T>(_);
+
+    /// Account states stored in this pallet
+    #[pallet::storage]
+    #[pallet::getter(fn account)]
+    pub type Account<T: Config> =
+        StorageMap<_, Blake2_128Concat, T::AccountId, AccountDataOf<T>, ValueQuery>;
+}
+
+impl<T: Config> StoredMap<T::AccountId, AccountDataOf<T>> for Pallet<T> {
```
Comment (markopoloparadox) [inline pallets/account-store/src/lib.rs:41]: Is it necessary to have tests for this pallet? :upside_down_face:

**Review by markopoloparadox — CHANGES_REQUESTED** [review]

File: types.json [hunk 2]
```diff
@@ -1,4 +1,5 @@
 {
+    "AccountDataOf": "AccountData",
```
Comment (markopoloparadox) [inline types.json:2]: Let's not forget to communicate with other teams that we yet again have changed the structure of the types file.

File: runtime/src/lib.rs [hunk 3]
```diff
@@ -84,10 +84,13 @@ construct_runtime!(
         Sudo: pallet_sudo::{Module, Call, Config<T>, Storage, Event<T>},
         System: frame_system::{Module, Call, Config, Storage, Event<T>},
         Treasury: pallet_treasury::{Module, Call, Storage, Config, Event<T>},
-        Timestamp: pallet_timestamp::{Module, Call, Storage, Inherent},
         TransactionPayment: pallet_transaction_payment::{Module, Storage},
         Utility: pallet_utility::{Module, Call, Storage, Event},
 
+        TiimeAccountStore: ternoa_account_store::{Module, Storage},
```
[the remark this answers is not in the document]
Comment (ETeissonniere) [reply runtime/src/lib.rs:90]: Go wonder :). Seems like that's the name that was chosen

File: types.json (hunk 2 above)
Comment (ETeissonniere) [reply types.json:2]: Very good point!

File: pallets/account-store/src/lib.rs (hunk 1 above)
Comment (ETeissonniere) [reply pallets/account-store/src/lib.rs:41]: Well there isn't much to test since we just proxy the calls to underlying `Account` storage type

Comment (ETeissonniere) [reply pallets/account-store/src/lib.rs:41]: As in, there isn't specific logic to really test - maybe we can add one simple unit test to check that things are correctly written though

**Review by markopoloparadox — APPROVED** [review]

File: pallets/account-store/src/lib.rs (hunk 1 above)
Comment (markopoloparadox) [reply pallets/account-store/src/lib.rs:41]: We can always do that later xd

**Review by ybensacq — APPROVED** [review]

**Pull request merged**